File: php/src/Scoreboard.php
<?php


namespace IceCream;


use DateTime;

class Scoreboard
{
    private Scorer $scorer;

    public function __construct(Scorer $scorer)
    {
        $this->scorer = $scorer;
    }

    public function printScoreboard(): void
    {
        $names = ["Steve","Julie","Francis"];

        echo "Scoreboard at time: " . (new DateTime())->format('r') . "\n";

        foreach ($names as $name) {
            $this->printScore($name);
        }
    }

    public function printScore(string $name): void
    {
        if ($name == "Steve") {
            $this->scorer->setFlavour('Strawberry');
        } else {
            $this->scorer->updateSelection();
        }
        $score = $this->scorer->getScore();
        echo "{$name} score: {$score}\n";
    }
}

File: C/src/scoreboard.c

#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <time.h>

#include "scoreboard.h"
#include "scorer.h"

void printScoreboard()
{
    const char *names[] = {"Steve","Julie","Francis"};
    time_t rawtime;
    struct tm * timeinfo;

    time ( &rawtime );
    timeinfo = localtime ( &rawtime );
    printf ( "Scoreboard at time: %s\n", asctime (timeinfo) );

    for (int i = 0; i < 3; ++i) {
        printScore(names[i]);
    }
}

void printScore(const char *name) {
    if (strcmp(name, "Steve") == 0)
    {
        flavour = strawberry;
    }
    else
    {
        updateSelection();  
    }
    int score = getScore();
    printf("%s score: %d\n", name, score);
}

